File: Backend-SpringBoot/tfgMaster/src/main/java/tfgMaster/entity/Rubrica.java
package tfgMaster.entity;

import java.util.Date;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToMany;
import jakarta.validation.constraints.NotBlank;

@Entity
public class Rubrica extends DomainEntity {
	@NotBlank
	private String descripcion;

	private boolean esBorrador;

	@JsonFormat(pattern = "dd-MM-yyyy")
	private Date fechaPublicacion;

	@ManyToMany
	private Set<Criterio> criterios;

	public Rubrica() {
		super();
	}

	public String getDescripcion() {
		return descripcion;
	}

	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	public boolean getEsBorrador() {
		return esBorrador;
	}

	public void setEsBorrador(boolean esBorrador) {
		this.esBorrador = esBorrador;
	}

	public Date getFechaPublicacion() {
		return fechaPublicacion;
	}

	public void setFechaPublicacion(Date fechaPublicacion) {
		this.fechaPublicacion = fechaPublicacion;
	}

	public Set<Criterio> getCriterios() {
		return criterios;
	}

	public void setCriterios(Set<Criterio> criterios) {
		this.criterios = criterios;
	}

}

File: Frontend-Angular/src/app/model/Rubrica.ts
import { Criterio } from "./Criterio";
import { Profesor } from "./Profesor";

export class Rubrica {
    id!: number;
    descripcion: string;
    esBorrador: boolean;
    fechaPublicacion: Date;
    criterios: Set<Criterio>;

    constructor(descripcion: string, esBorrador: boolean, fechaPublicacion: Date, criterios: Set<Criterio>
    ) {
        this.descripcion = descripcion,
            this.esBorrador = esBorrador,
            this.fechaPublicacion = fechaPublicacion,
            this.criterios = criterios
    }
}